### ISSUE TITLE
Problem on starting the repl: "REPL server launch timed out"

### ISSUE THREAD

**Issue by alexandru** [magomimmo/cljs-start#10]
The output:

``` bash
$ lein repl
Compiling ClojureScript.
Compiling "dev-resources/public/js/advanced.js" from ["src/cljs" "test/cljs"]...
Successfully compiled "dev-resources/public/js/advanced.js" in 12.255468837 seconds.
Compiling "dev-resources/public/js/simple.js" from ["src/cljs" "test/cljs"]...
Successfully compiled "dev-resources/public/js/simple.js" in 8.628436487 seconds.
Compiling "dev-resources/public/js/piratebadge.js" from ["src/cljs" "test/cljs" "dev-resources/tools/repl"]...
Successfully compiled "dev-resources/public/js/piratebadge.js" in 21.962943446 seconds.
Compiling "dev-resources/public/js/useless.js" from ["src/cljs"]...
Successfully compiled "dev-resources/public/js/useless.js" in 5.207148103 seconds.
REPL server launch timed out.
```

**Comment by magomimmo** [comment]
Yeah, it depends on the fact that the compilation process is very long and the REPL server should have a timeout which is set to a value that it's not enough for the compilation process to reach its end.

You have two options:

a) issue `lein compile` before issuing `lein repl`: this way all the very long compilations have been already executed
b) relaunch `lein repl`: this way some long compilations have already been executed.

I suggest a)

HIH

**Comment by magomimmo** [comment]
I'm on the way to publish a release which, by using profiles even more deeply that now, is able to workaround the compilation time required to compile the builds all together.

**Comment by alexandru** [comment]
Thanks,

The compilation times varies depending on how busy my laptop is. However, given that starting a project means adding code to it :-) those compilation times will only increase, so it's a problem because that timeout period will be reached eventually.

But btw, I'm a rookie so I don't know - shouldn't nRepl have a parameter for specifying or disabling that timeout?

**Comment by magomimmo** [comment]
Thanks to a suggestion from Laurent Petit I now have a solution which uses lein profiles.

Right now I'm testing it and in a couple of our it will be published. 

mimmo

**Comment by magomimmo** [comment]
just published the cljs-start 0.1.1 release which significant reduces the compilation time by defining different lein profiles (:dev, :simple and :advanced).

try the following:

lein new cljs-start wonderful-lib
cd wonderful-lib
lein compile
lein repl
(run)
(browser-repl)

visit the localhost:3000 an wait a little bit for the connection to be established before repling

To compile and run test for the :simple to :advanced compilation

lein with-profiles simple do clean, compile, test

HIH

mimmo

**Comment by magomimmo** [comment]
I close the issue, which is now solved with the 0.1.1 release.
